### DISCUSSION TITLE
What version of SpaNet controllers is this tested with?

### DISCUSSION THREAD

**Discussion by mattbnz** [wayne-love/sn_esp32#8] [Q&A] [answered]
I'm having a real challenging time getting a serial connection up and running - to the point where I'm wondering if my SpaNet controller is new and something has changed... would you mind sharing the version of the controller that you're working with so I can eliminate that? 

My controller has SV-MINI PCB V1.3.1 9 1 17 stencilled on the PCB 

Plugging in to EXP1, I get the 12V/GND fine on pins 7/8, but no matter what I connect to or send from pins 5/6 I cannot get any serial communication going, I've tried:
- wemos D1-mini boards
- USB-BUB serial convertor
- Bus Pirate in UART mode

With the first two, sending RF\n, gets back a few bytes of garbage (all bits set, etc). 

The bus pirate reads a continual stream of zero byte frames, (but with framing errors marked - e.g. as if the BAUD is wrong), if I send RF\n via the bus pirate then I get two bogus frames and one good frame but with bogus data back... it really feels like whatever is at the other end of EXP1 is not wanting to communicate at 38400 8N1...

UART>0x52,0x46,0x0A
WRITE: 0x52 
WRITE: 0x46 
WRITE: 0x0A 
UART>
READ: -f 0x2B

READ: -f 0x01

READ: 0xFD
UART>
READ: -f 0x00

I've tried all the other standard baud rates on the off chance my controller uses a different rate, but no joy.

So I'm stumped, if you can let me know what you've got it working with, that would be most useful! Thanks.

**Comment by mattbnz** [comment]
It looks like I've actually bricked my controller (keypad doesn't function, HB Led is doing a double-flash implying the heater controller is not present/reporting), I suspect I've shorted something in the section for the 12V accessory ports by mistake while hooking it up unfortunately :( 

This might be an expensive lesson...

**Comment by wayne-love** [comment, marked as answer]
@mattbnz do you have a common ground for all devices?  I found out the hard way, not having a common ground reference voltage leads just noise on the rx/tx lines, this wasted several days of my time during the discovery phase.  The symptoms of the problem are very similar to what you are encountering.

Re compatability, this should work for all controllers that can be connected to the spanet smartlink box.  we're using exactly the same pins and commands as the genuine wifi connector.  My spa is sunken into the ground and access is a bitch, the next time that I'm under there I will check the HW version.

Re bricking the controller - Bugger! - Condolences.  Have you tried a factory reset?

**Reply by mattbnz** [reply]
I had power for the Wemos supplied from the Spa, so I think that should be sufficient for common ground - no other power supplies in use. 

Unless there is a way to factory reset from the circuit board itself, I don't think I can... I've tried the procedure from https://spanet.com.au/perform-eprm-software-reset/ but there's absolutely zero response from the keypad. Looking at the HB led on the board inside the controller it's doing the on, on, off sequence for ER-2 Heater Plug (no heat sensor communication), which of course plugs into the same row of connectors in the 12V section, so I'm suspecting that I've broken that bus/module entirely.

Thanks for confirming re versions - you're right re the official connector - it's marked as compatible with my Spa, so of course the idea that it's a newer revision of the controller and that was my problem was a vain hope :)

Cheers.

**Comment by JonathanGiles** [comment]
@mattbnz I'm keen to hear how you got on? Did you brick your controller? Did you get things working?

**Comment by mattbnz** [comment]
Yes, my controller is bricked. I ended up having to buy a new one (NZD$1k) to get the spa working again. The old/bricked one is sitting in the corner of my office waiting for me to have time to try and reverse engineer/fix it.

**Reply by JonathanGiles** [reply]
Ouch, sorry to hear that! Do you know what caused it? It sounds like you miswired things?

**Comment by mattbnz** [comment]
AFAIK I never miswired it, BUT given the symptoms, sending 12V up the wrong wires does seem like the most likely explanation.

**Reply by vdbpeter** [reply]
Gidday @mattbnz , I'm attempting to get my Spanet SV3 Exp1 port active, but not getting any responses from the device. After reading the various setups and setting the port speed manually in RealTerm, sending "RF" with CR & LF (38k & 115k) and getting no responses, I wrote a script that would send out a "RF\r\n" and waits for "RF" that looped through the common speeds from 4.8k through to 2M. Still nothing.

 I'm hoping you could confirm a thing or 2 about the physical interface, please?

I've done some volt & ohm metering, and found this on my wiring against Pin7 (that's reported as Gnd):
pin:
1 +14v
2 0.2v (noise?)
3 0.2v (noise?)
4 0.2v (noise?)
5 3.3v (Rx?)
6 4.9v (Tx?)
7 0v (Ground- but isolated from common ground- no connection to steel frame)
8 No measurement readable.

Seeing your units are relatively new, it's likely to be a similar hardware version to mine (SpaNET SV BRAIN V5.1 4/9/12, Firmware: sv-t43 6.21-06-02). Can you confirm?

Talking to my supplier, to add a Wifi unit to my pool requires no setup- plug & play. However, in the Setup Menu, there's a Wifi menu with the options: Reset, Infra or Hot. (He says they dont change anything). Selecting these makes no difference. 

So, did you get your replacement working with the microcontroller on Exp1? If so, what did you have to do to get it working?

And can you confirm the physical interface wiring, please?

Thanks

**Comment by wayne-love** [comment]
Sounds like you don’t have a common ground between the spa and the device that you are using to read.Baud is 115200.Always send a \n and wait (about 200ms) before sending the real command to flush any noise from the buffer.IIRC a \r is not required.Your supplier is right, there is nothing to change on the spa menus.I’m travelling at the moment so cannot confirm the pin outs but you seem close.  1/2 and 7/8 are either a 12v signal or 12v power, I can’t remember which is which but if you have a look at my circuit it will give you the required pins.On 16 Jan 2023, at 10:26 am, vdbpeter ***@***.***> wrote:﻿
Gidday @mattbnz , I'm attempting to get my Spanet SV3 Exp1 port active, but not getting any responses from the device. After reading the various setups and setting the port speed manually in RealTerm, sending "RF" with CR & LF (38k & 115k) and getting no responses, I wrote a script that would send out a "RF\r\n" and waits for "RF" that looped through the common speeds from 4.8k through to 2M. Still nothing.
I'm hoping you could confirm a thing or 2 about the physical interface, please?
I've done some volt & ohm metering, and found this on my wiring against Pin7 (that's reported as Gnd):
pin:
1 +14v
2 0.2v (noise?)
3 0.2v (noise?)
4 0.2v (noise?)
5 3.3v (Rx?)
6 4.9v (Tx?)
7 0v (Ground- but isolated from common ground- no connection to steel frame)
8 No measurement readable.
Seeing your units are relatively new, it's likely to be a similar hardware version to mine (SpaNET SV BRAIN V5.1 4/9/12, Firmware: sv-t43 6.21-06-02). Can you confirm?
Talking to my supplier, to add a Wifi unit to my pool requires no setup- plug & play. However, in the Setup Menu, there's a Wifi menu with the options: Reset, Infra or Hot. (He says they dont change anything). Selecting these makes no difference.
So, did you get your replacement working with the microcontroller on Exp1? If so, what did you have to do to get it working?
And can you confirm the physical interface wiring, please?
Thanks

**Reply by vdbpeter** [reply]
Got it working! (by carefully reading the experiences in the discussion on https://gist.github.com/devbobo/b82134291d96f384137e01f39a7f2a01 Lots of great information including photos of reverse engineering attempts)

So, the few key things I picked up that helped me get it working (and hopefully future others):

My unit: SpaNET SV3, SV BRAIN V5.1 4/9/12, Firmware: sv-t43 6.21-06-02

Used "RealTerm" to test commands with USB-RS232 adapter.

- Pinout of my Exp1 (no other peripheral was plugged in): 
Pin 1 +14v
Pin 5 Tx (3.3v idle) - connect to terminal device Rx
Pin 6 Rx (4.9v idle) - connect to terminal device Tx
Pin 7 Signal Ground - NOT EARTH- isolated from common ground- no connection to internal steel frame.

- speed 38400 bps
- Send command with ONLY '\n', eg; 'RF\n' (no quotes). If a CR is added, the command is ignored.

Now I will progress integrating it into my network.

(Maybe we collate people's hardware configurations into a table, seeing there are some differences?)

**Comment by wayne-love** [comment]
Just want to correct myself.  Baud is definitely 38400, not 115200 mentioned in previous comments.
